torch/nn/modules/module.py

```diff
@@ -967,10 +967,12 @@ def _load_from_state_dict(self, state_dict, prefix, local_metadata, strict,
                 input_param = state_dict[key]
 
                 # Backward compatibility: loading 1-dim tensor from 0.3.* to version 0.4+
-                if len(param.shape) == 0 and len(input_param.shape) == 1:
+                # We access data.shape because for uninitialized parameters `param.shape`
+                # is forbidden
+                if len(param.data.shape) == 0 and len(input_param.data.shape) == 1:
```

Comment:
I know this is "ok" because these are `Parameter` but I really don't like to see `.data` added here.
Given the current code, you could use `.size()` or `.ndim()` that you don't block.

Reply:
I think it's actually fine to just do `isinstance(param, UninitializedParam)` here, it justifies intent much better and I don't think adds too much cyclical dependency